library/src/lib/time/time.component.ts

```diff
@@ -256,97 +290,85 @@ export class TimeComponent implements OnChanges, ControlValueAccessor {
     }
 
     /** Toggles the period (am/pm). */
-    togglePeriod() {
+    togglePeriod(): void {
         if (this.time.hour < 24 && this.time.hour >= 0) {
-            if (this.period === 'am') {
-                this.period = 'pm';
+            if (this.isAm(this.period)) {
+                this.period = this.timeI18n.meridianPm;
                 this.periodModelChange();
-            } else if (this.period === 'pm') {
-                this.period = 'am';
+            } else if (this.isPm(this.period)) {
+                this.period = this.timeI18n.meridianAm;
                 this.periodModelChange();
             }
         }
     }
 
-    /** @hidden */
-    hourModelChange() {
-        if (this.meridian) {
-            if (!(this.time.hour > 12 || this.time.hour < 0) || !this.validate) {
-                this.onChange(this.time);
-            }
-        } else {
-            if (!(this.time.hour > 23 || this.time.hour < 0) || !this.validate) {
-                this.onChange(this.time);
-            }
-        }
-    }
-
-    /** @hidden */
-    minuteModelChange() {
+    /** @hidden
+     * Handles minutes model change from template
+     * */
+    minuteModelChange(): void {
         if (!(this.time.minute > 59 || this.time.minute < 0) || !this.validate) {
             this.onChange(this.time);
         }
     }
 
-    /** @hidden */
-    secondModelChange() {
+    /** @hidden
+     * Handles seconds model change from template
+     * */
+    secondModelChange(): void {
         if (!(this.time.second > 59 || this.time.second < 0) || !this.validate) {
             this.onChange(this.time);
         }
     }
 
-    /** @hidden */
-    periodModelChange() {
-        this.period = this.period.toLowerCase();
-        if (this.period !== 'am' && this.period !== 'pm') {
+    /** @hidden
+     * Handles period model change. depending on current hour and new period changes hours by +/- 12
+     */
+    periodModelChange(): void {
+        if (this.time && !this.time.hour) {
+            this.time.hour = 0;
+        }
+        if (!this.period ||
+            (!this.isPm(this.period) && !this.isAm(this.period))
+        ) {
             this.periodInvalid = true;
```

Comment:
should we reset the invalid input like we do with hours/minutes/secs? 
![Screen Shot 2019-08-08 at 4 16 46 PM](https://user-images.githubusercontent.com/39598672/62734888-021d1880-b9f8-11e9-8ea6-379ab8efa0af.png)

Reply:
It's very easy to achieve, but I'm not sure if that's how it's supposed to be.

Reply:
Actually it will improve UX, I'm gonna try it.